### DISCUSSION TITLE
Welcome to laravel-g4t-swagger-auto-generate Discussions!

### DISCUSSION THREAD

**Discussion by hussein4alaa** [hussein4alaa/laravel-g4t-swagger-auto-generate#13] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by IsaMohamud** [comment]
Hello hussein,
I was wondering if it is possible to change the auto generated swagger api documentation.
In my case there are some endpoints not correctly documented, which causes errors in the execute state.

**Reply by hussein4alaa** [reply]
Hello IsaMohamud,
Welcome
Can you provide me the routes code from your routes file?
And if you can give me the error messages that you see

**Comment by IsaMohamud** [comment]
![swagger-error](https://github.com/hussein4alaa/laravel-g4t-swagger-auto-generate/assets/78957849/1d145fd4-028c-48ff-97e1-b027f0af4289)

In the above image you can see the error message, which indicates that certain fields are missing. This is because the api request body is structured as follows:
{
  "onderwijsprogramma.uuid": "3fa85f64-5717-4562-b3fc-2c963f66afa6",
  "onderwijsprogramma.naam": "string",
  "onderwijsprogramma.code": "string",
  "onderwijsprogramma.cohort": "string",
  "onderwijsprogramma.niveau": "string",
  "onderwijsprogramma.uitvoeringsvorm": "string",
  "onderwijsprogramma.duur": "string",
  "onderwijsprogramma.duur_eenheid": "string",
  "onderwijsprogramma.indicatie": "string",
  "onderwijsprogramma.programma_type": "string",
  "versie.naam": "string",
  "versie.omschrijving": "string",
  "versie.commerciele_naam": "string",
  "versie.startdatum": "2023-10-09",
  "versie.einddatum": "2023-10-09"
}

However the controller method of the route being invoked expects the following format:

{
  "onderwijsprogramma": {
    "uuid": "152bc989-33f3-373f-81bf-5f9f188ef2b1",
    "naam": "voluptatem",
    "code": "porro",
    "cohort": "blanditiis",
    "niveau": "quo",
    "uitvoeringsvorm": "odio",
    "duur": 20,
    "duur_eenheid": "fugit",
    "indicatie": "ipsam",
    "programma_type": "rerum",
  },
  "versie": {
    "naam": "delectus",
    "omschrijving": "omnis",
    "commerciele_naam": "nisi",
    "startdatum": "2023-09-28T08:48:57",
    "einddatum": "2023-09-28T08:48:57"
  }
}

I think it doesn't have to do with something in the routes. That is just a simple code to create an apiResource.
Like this: Route::apiResource('onderwijsprogrammas', OnderwijsprogrammasController::class);

**Reply by hussein4alaa** [reply]
I understand what your problem is and will work to solve it asap.

Thank you

**Reply by IsaMohamud** [reply]
Thank you for your fast responses, i am looking forward to hear from you if it is done.

**Reply by hussein4alaa** [reply]
@IsaMohamud Now you can install new version 2.0.2 `composer require g4t/swagger:2.0.2` 
Nested input issue was fixed
I look forward to hearing your thoughts and feedback

**Reply by IsaMohamud** [reply]
Unfortunately it doesn't work. If i install the new version my whole composer.json file crashes.
It says that Swagger properties needs to be added in the composer file.

**Reply by hussein4alaa** [reply]
@IsaMohamud you can remove composer.lock file
and replace old version in composer.json to new version `"g4t/swagger": "2.0.2",`
and run `composer update`
![Screenshot 2023-10-09 at 6 32 32 PM](https://github.com/hussein4alaa/laravel-g4t-swagger-auto-generate/assets/13614747/36524051-f860-42bc-8437-aa6d7157ccc5)

**Reply by IsaMohamud** [reply]
The new version is now successfully installed. But now i got the error below.
![image](https://github.com/hussein4alaa/laravel-g4t-swagger-auto-generate/assets/78957849/16198f1b-efe6-4edd-bb7e-8a36344b4517)

**Reply by hussein4alaa** [reply]
Hi @IsaMohamud 

Thank you so much for taking the time to discover and report these issues! Your feedback is incredibly valuable, and I genuinely appreciate your help in making our package better.

I want to let you know that I've been hard at work addressing the problems you pointed out.
I believe I've fixed everything you mentioned. I'd be grateful if you could take some time to test the new version and let me know how it works for you.
Your continued feedback will help ensure the package's quality and reliability.

Once again, thank you for your contributions, and please feel free to reach out if you encounter any further issues or have any more feedback to share.

Best regards,

**Reply by hussein4alaa** [reply]
please use "g4t/swagger": "2.0.5",

**Reply by IsaMohamud** [reply]
No problem, you are welcome. I see the potential in this package, so i don't mind to invest some time to it.

The next problem i got, has to do with reaching the swagger page /documentation/swagger.
See below image.
![image](https://github.com/hussein4alaa/laravel-g4t-swagger-auto-generate/assets/78957849/4be9a3a9-21d7-48bd-a0b4-ab62940d007a)

**Reply by hussein4alaa** [reply]
i think this issue in your middleware

**Reply by IsaMohamud** [reply]
I don't think that is the problem. Nothing is changed in the code and i only updated to a new version of your library.
With the older version i don't have this issue. 

It has something to do with the response that is sent with the endpoint /swagger/documentation. My middleware can't handle it.

**Reply by IsaMohamud** [reply]
![image](https://github.com/hussein4alaa/laravel-g4t-swagger-auto-generate/assets/78957849/65cf92bd-281a-413c-abe3-f5234daa6e28)
In the above image you can see the last item of the error stack. Maybe it is something because of the double [] that cannot be used
on a string?

**Comment by IsaMohamud** [comment]
Hello hussein,

Do you already have a solution for the problem? I still think the problem has something to do with the function generateSchemaNested,
where the following line of code gives an error: $current = &$current[$nestedKey]['properties'];
